jms/src/test/java/akka/stream/alpakka/jms/javadsl/JmsConnectorsTest.java

```diff
@@ -299,6 +301,51 @@ public void publishAndConsumeTopic() throws Exception {
         });
     }
 
+    @Test
+    public void publishAndConsumeJmsTextMessagesWithClientAcknowledgement() throws Exception {
+        withServer(ctx -> {
+            ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory(ctx.url);
+
+            Sink<JmsTextMessage, NotUsed> jmsSink = JmsSink.create(
+                    JmsSinkSettings
+                            .create(connectionFactory)
+                            .withQueue("test")
+            );
+            List<Integer> intsIn = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+            List<JmsTextMessage> msgsIn = new ArrayList<>();
+            for(Integer n: intsIn) {
+                msgsIn.add(JmsTextMessage.create(n.toString()));
+            }
+
+            Source.from(msgsIn).runWith(jmsSink, materializer);
+
+            //#create-jms-source-client-ack
+            Source<Message, NotUsed> jmsSource = JmsSource.create(JmsSourceSettings
+                    .create(connectionFactory)
+                    .withQueue("test")
+                    .withAcknowledgeMode(ClientAcknowledge$.MODULE$)
```

Comment:
This is not ok as a Java API, I think my proposed change will fix it though. If it doesn't, put the constants in a non-companion object called `AcknowledgeModes.ClientAck`

Reply:
@johanandren I have implemented the final class + compation object variant, please take a look how java api looks now